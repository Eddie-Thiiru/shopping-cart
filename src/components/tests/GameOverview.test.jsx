import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Overview from "../GameOverview";

const data = {
  data: {
    imageURL: "https://example.com/api/product/1111",
    name: "Grand Theft Auto 5",
    metaScore: 92,
    genre: [{ name: "Action" }, { name: "Adventure" }],
    platforms: [
      { platform: { name: "PC" } },
      { platform: { name: "PlayStation" } },
      { platform: { name: "XBOX" } },
    ],
    released: "march 15, 2013",
    price: 29.99,
  },
};

describe("Game overview", () => {
  it("renders overview while using useLocation data", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[{ state: data }]}>
        <Overview />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
