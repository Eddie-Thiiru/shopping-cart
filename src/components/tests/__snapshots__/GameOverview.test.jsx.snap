// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game overview > renders overview while using useLocation data 1`] = `
<div>
  <div
    class="gameOverview"
  >
    <div
      class="gameDetails"
    >
      <img
        alt="Grand Theft Auto 5 background image"
        src="https://example.com/api/product/1111"
      />
      <div
        class="gameAbout"
      >
        <p />
      </div>
      <div
        class="gameMeta"
      >
        <div
          class="gameGenreScore"
        >
          <div
            class="genre"
          >
            <h4>
              Genre
            </h4>
            <p>
              Action 
              Adventure 
               
            </p>
          </div>
          <div
            class="metaScore"
          >
            <h4>
              Metascore
            </h4>
            <p>
              92
            </p>
          </div>
        </div>
        <div
          class="releaseDate"
        >
          <h4>
            Release Date
          </h4>
          <p>
            march 15, 2013
          </p>
        </div>
        <div
          class="gamePlatforms"
        >
          <p>
            PC 
            PlayStation 
            XBOX 
          </p>
        </div>
      </div>
    </div>
    <div
      class="shopSection"
    >
      <h2>
        Grand Theft Auto 5
      </h2>
      <p>
        29.99
      </p>
      <div>
        <button
          type="button"
        >
          ADD TO CART
        </button>
        <button
          type="button"
        >
          BUY NOW
        </button>
      </div>
    </div>
  </div>
</div>
`;
